src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,37 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(animal) {
+    this.name = animal;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(animal, health) {
+    super(animal, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(predator) {
+    if (predator.hidden === false) {
+      predator.health -= 50;
+    }
+
+    if (predator.health === 0) {
```

Comment:
good solution 4 my point of view, cos array will filtering only when animal health is 0, not after every bite 👍